src/node_contextify.cc

```diff
@@ -745,6 +747,26 @@ class ContextifyScript : public BaseObject {
   }
 
 
+  static void CreateCachedData(const FunctionCallbackInfo<Value>& args) {
+    Environment* env = Environment::GetCurrent(args);
+    ContextifyScript* wrapped_script;
+    ASSIGN_OR_RETURN_UNWRAP(&wrapped_script, args.Holder());
+    Local<UnboundScript> unbound_script =
+        PersistentToLocal(env->isolate(), wrapped_script->script_);
+    std::unique_ptr<ScriptCompiler::CachedData> cached_data(
+        ScriptCompiler::CreateCodeCache(unbound_script));
+    if (!cached_data) {
+      THROW_ERR_CREATE_CACHED_DATA_FAILED(env, "Creating cached data failed.");
```

Comment:
I’m a bit torn on this … it’s a *cache*, so it feels like it should fail “softly” … maybe in the case of a failure we could just return an empty buffer instead?

Reply:
since its an explicit call i think its best to throw

Reply:
You could always return `null` in the case of no cache being able to be generated. It's easy to detect and doesn't require wrapping the method in a try-catch.

Reply:
we could return null in lots of places throughout core but we use errors to signal that something went wrong... i don't understand the aversion here.

Reply:
> i don't understand the aversion here.

I come from user-land which tries to smooth over these kinds of errors _(I lean that that)_.
If I get a cache I'll likely store it somewhere, but if I don't then I won't. The current API (before this) doesn't error if the cache isn't produced so I dig the non-erroring route is all.

Reply:
Is it possible for V8 to provide failure reasons in the future? If so we could attach the reason to the error object which might be useful to users. Or we can check in advance, right now the reasons seem to be

```cpp
  if (script->ContainsAsmModule()) return nullptr;
  if (isolate->debug()->is_loaded()) return nullptr;
```

Reply:
I'm also +1 to throwing.